contracts/InverseVester.sol

```diff
@@ -7,78 +7,101 @@ import "./interfaces/IInv.sol";
 /**
  * @title Inverse token vesting contract
  * @author Inverse Finance
- * @notice Base contract for vesting agreement on INV tokens
- *         Reverse vesting means tokens are owned but the recipient
- *         and can be used for voting.
+ * @notice Contract for vesting agreement on INV tokens
  * @dev    Vesting calculation is linear
  */
 contract InverseVester is Ownable {
     using SafeERC20 for IInv;
 
     uint256 public constant DAY = 1 days;
 
-    // @dev Timestamp for the start of this vesting agreement
+    /// @dev Timestamp for the start of this vesting agreement
     uint256 public vestingBegin;
 
-    // @dev Timestamp for the end of this vesting agreement
+    /// @dev Timestamp for the end of this vesting agreement
     uint256 public vestingEnd;
 
-    // @dev Timestamp for the last time vested tokens were claimed
+    /// @dev Timestamp for the last time vested tokens were claimed
     uint256 public lastClaimTimestamp;
 
-    // @dev Total amount to be vested
+    /// @dev Total amount to be vested
     uint256 public immutable vestingAmount;
 
-    // @dev Inverse finance treasury token
-    IInv public immutable inv;
+    /// @dev By how long the vesting should be delayed after activating this contract
+    ///      This can be used for multi year vesting agreements
+    uint16 public immutable vestingStartDelayInDays;
```

Comment:
With the timelock in charge of deploying new vesting contracts, I imagine it can immediately transfer `INV` into the contract after creating it, and afterwards set the start date of vesting. Wouldn't that make the delay obsolete?

I imagine that would also allow most of the `activate` function to be moved into the constructor, with the rest being no longer needed.

The only downside I see is that the entirety of the process then resides at the side of the treasury, effectively giving the recipient no choice code-wise. But, and now I'm just writing down whatever I'm thinking, that is already the case right now, but the recipient has the illusion of setting a starting date for vesting. In pretty much every case for the current vesting contract I feel like the `recipient` would immediately call `activate` anyway.

Reply:
Not really. 
In my use case I need:
1) To be able to activate at an unknow date
2) To activate a contract which vesting starts 1 year in the future but gives me votes at the point of activation